tests/queries/0_stateless/01452_replicated_drop_part.sql

```diff
@@ -0,0 +1,20 @@
+SET replication_alter_partitions_sync = 2;
+
+DROP TABLE IF EXISTS attach_r1;
+DROP TABLE IF EXISTS attach_r2;
+
+CREATE TABLE attach_r1 (v UInt8) ENGINE = ReplicatedMergeTree('/clickhouse/tables/test/01452/attach', 'r1') order by tuple() settings max_replicated_merges_in_queue = 0;
+CREATE TABLE attach_r2 (v UInt8) ENGINE = ReplicatedMergeTree('/clickhouse/tables/test/01452/attach', 'r2') order by tuple() settings max_replicated_merges_in_queue = 0;
+
+INSERT INTO attach_r1 VALUES (0);
```

Comment:
Also need to SYSTEM STOP MERGES before INSERT (and start after ALTER), otherwise parts can be merged before ALTER.

Reply:
@alexey-milovidov i'm using `settings max_replicated_merges_in_queue = 0;` setting on the table to prevent merges being scheduled. `STOP MERGES` prevents actual merges from happening but if a merge is even scheduled then `DROP PART` will fail.